packages/web-app-files/tests/unit/components/SideBar/SideBar.spec.ts

```diff
@@ -0,0 +1,126 @@
+import fileSideBars from 'web-app-files/src/fileSideBars'
+import InnerSideBar from 'web-pkg/src/components/sideBar/SideBar.vue'
+import SideBar from 'web-app-files/src/components/SideBar/SideBar.vue'
+import { Resource } from 'web-client/src/helpers'
+import { mockDeep } from 'jest-mock-extended'
+import {
+  createStore,
+  defaultComponentMocks,
+  defaultPlugins,
+  defaultStubs,
+  shallowMount
+} from 'web-test-helpers'
+import { defaultStoreMockOptions } from 'web-test-helpers/src/mocks/store/defaultStoreMockOptions'
+
+jest.mock('web-pkg/src/observer')
+jest.mock('web-app-files/src/helpers/resources', () => {
+  const original = jest.requireActual('web-app-files/src/helpers/resources')
+  return {
+    ...original,
+    buildResource: jest.fn()
+  }
+})
+
+const selectors = {
+  noSelectionInfoPanel: 'noselection-stub'
+}
+
+describe('SideBar', () => {
+  describe('no selection info panel', () => {
+    afterEach(() => {
+      jest.clearAllMocks()
+    })
+    describe('for public links', () => {
+      it.each([
+        [
+          'shows in root node',
+          {
+            path: '',
+            noSelectionExpected: true
+          }
+        ],
+        [
+          'does not show in non-root node',
+          {
+            path: '/publicLinkToken/some-folder',
+            noSelectionExpected: false
+          }
+        ]
+      ])('%s', async (name, { path, noSelectionExpected }) => {
+        const item = mockDeep<Resource>({ path })
+        const { wrapper } = createWrapper({ item })
+        await wrapper.vm.$nextTick()
+        await wrapper.vm.$nextTick()
```

Comment:
Is this on twice on purpose?

Reply:
Yes, that's pretty ugly - but we unfortunately need it, when we have a chain of promises. 

As Promises are always async (even if they return synchronously), we need a `nextTick` for every `.then` :-(
Can hopefully be solved with something like `allPromisesSettled` from the/a test framework - but I would recommend to tackle that in a separate PR...

Reply:
Maybe this https://jestjs.io/docs/jest-object#jestrunallticks

Reply:
Nope, doesn't work ... there's a lengthy thread in the jest repo without any real conclusion: https://github.com/facebook/jest/issues/2157